pkg/controller/defaultdomaincert/default_domain_cert_controller.go

```diff
@@ -0,0 +1,154 @@
+	"github.com/Azure/aks-app-routing-operator/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var name = controllername.New("default", "domain", "cert", "reconciler")
+
+type defaultDomainCertControllerReconciler struct {
+	client client.Client
+	events record.EventRecorder
+	conf   *config.Config
+	store  store.Store
+}
+
+func NewReconciler(conf *config.Config, mgr ctrl.Manager, store store.Store) error {
+	metrics.InitControllerMetrics(name)
+
+	if err := store.AddFile(conf.DefaultDomainCertPath); err != nil {
+		return fmt.Errorf("adding default domain cert %s to store: %w", conf.DefaultDomainCertPath, err)
+	}
+
+	if err := store.AddFile(conf.DefaultDomainKeyPath); err != nil {
+		return fmt.Errorf("adding default domain key %s to store: %w", conf.DefaultDomainKeyPath, err)
+	}
+
+	reconciler := &defaultDomainCertControllerReconciler{
+		client: mgr.GetClient(),
+		events: mgr.GetEventRecorderFor("aks-app-routing-operator"),
+		conf:   conf,
+		store:  store,
+	}
+
+	if err := name.AddToController(
+		ctrl.NewControllerManagedBy(mgr).
+			For(&approutingv1alpha1.DefaultDomainCertificate{}).
+			Owns(&corev1.Secret{}),
+		mgr.GetLogger(),
+	).Complete(reconciler); err != nil {
+		return fmt.Errorf("building the controller: %w", err)
+	}
+
+	return nil
+}
+
+func (d *defaultDomainCertControllerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
+	start := time.Now()
+	lgr := log.FromContext(ctx, "name", req.Name, "namespace", req.Namespace)
+	ctx = log.IntoContext(ctx, lgr)
+
+	lgr.Info("reconciling DefaultDomainCertificate")
+	defer func() {
+		lgr.Info("reconcile finished", "latency", time.Since(start))
+	}()
+
+	defaultDomainCertificate := &approutingv1alpha1.DefaultDomainCertificate{}
+	if err := d.client.Get(ctx, req.NamespacedName, defaultDomainCertificate); err != nil {
+		if apierrors.IsNotFound(err) { // object was deleted
+			lgr.Info("DefaultDomainCertificate not found")
+			return ctrl.Result{}, nil
+		}
+
+		lgr.Error(err, "unable to get DefaultDomainCertificate")
+		return ctrl.Result{}, err
+	}
+
+	if defaultDomainCertificate.Spec.Target.Secret == nil {
+		err := errors.New("DefaultDomainCertificate has no target secret specified")
+		lgr.Error(err, "DefaultDomainCertificate has no target specified, this should be blocked by CRD CEL validation")
+		return ctrl.Result{}, err
+	}
+
+	lgr = lgr.WithValues("secretTarget", *defaultDomainCertificate.Spec.Target.Secret)
+	ctx = log.IntoContext(ctx, lgr)
+
+	lgr.Info("upserting Secret for DefaultDomainCertificate")
+	secret, err := d.getSecret(defaultDomainCertificate)
+	if err != nil {
+		err := fmt.Errorf("getting Secret for DefaultDomainCertificate: %w", err)
```

Comment:
generating vs getting? get to me implies that it already exists

Reply:
changed to generating